shared/actions/chat.js

```diff
@@ -1143,16 +1143,26 @@ function * _badgeAppForChat (action: BadgeAppForChat): SagaGenerator<any, any> {
   const selectedConversationIDKey = yield select(_selectedSelector)
   const windowFocused = yield select(_focusedSelector)
 
-  const newConversations = _.reduce(conversations, (acc, conv) => {
+  const newConversations = conversations.reduce((acc, conv) => {
     // Badge this conversation if it's unread and either the app doesn't have
     // focus (so the user didn't see the message) or the conversation isn't
     // selected (same).
-    const unread = conv.UnreadMessages > 0
-    const selected = (conversationIDToKey(conv.convID) === selectedConversationIDKey)
+    const unread = conv.get('UnreadMessages') > 0
+    const selected = (conversationIDToKey(conv.get('convID')) === selectedConversationIDKey)
     const addThisConv = (unread && (!selected || !windowFocused))
     return addThisConv ? acc + 1 : acc
   }, 0)
   yield put(badgeApp('chatInbox', newConversations > 0, newConversations))
+
+  let conversationsWithKeys = {}
+  conversations.map(conv => {
+    conversationsWithKeys[conversationIDToKey(conv.get('convID'))] = conv.get('UnreadMessages')
+  })
+  const conversationUnreadCounts = Map(conversationsWithKeys)
+  yield put({
+    payload: conversationUnreadCounts,
+    type: 'chat:updateConversationUnreadCounts',
+  })
 }
```

Comment:
let's leave this for now. i'm going to move this into the converstaionStates in my next pr, seems like it belongs there